src/components/Reports/GDPR/SubscriberGdpr.js

```diff
@@ -22,20 +22,34 @@ const SubscriberGdpr = ({ location, dependencies: { dopplerApiClient } }) => {
       setState({ loading: true });
       const email = replaceSpaceWithSigns(extractParameter(location, queryString.parse, 'email'));
       const responseSubscriber = await dopplerApiClient.getSubscriber(email);
-      if (responseSubscriber.success) {
+      const allFields = await dopplerApiClient.getUserFields();
+      if (responseSubscriber.success && allFields.success) {
         const subscriber = {
           firstName: responseSubscriber.value.fields.find((x) => x.name === 'FIRSTNAME'),
           lastName: responseSubscriber.value.fields.find((x) => x.name === 'LASTNAME'),
           email: responseSubscriber.value.email,
           score: responseSubscriber.value.score,
           status: responseSubscriber.value.status,
         };
+        const allPermissionFields = allFields.value.filter(
+          (customField) => customField.type === 'permission' || customField.type === 'consent',
+        );
+        const subscriberPermissionFields = responseSubscriber.value.fields.filter(
+          (customField) => customField.type === 'permission' || customField.type === 'consent',
+        );
+
+        const fields = allPermissionFields.map((field) => {
+          const found = subscriberPermissionFields.find(
+            (subscriberField) => subscriberField.name === field.name,
+          );
+          field.value = found ? found.value : 'none';
+          return field;
+        });
```

Comment:
dont know if works but is another option :)
` return subscriberPermissionFields.find(
            (subscriberField) => subscriberField.name === field.name,
          ).value || 'none' ;

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find array.find returns `undefined` if no item is found.